test/distributed/org/apache/cassandra/fuzz/harry/integration/dsl/HistoryBuilderOverridesTest.java

```diff
@@ -0,0 +1,376 @@
+                                       (e) -> schema.ckGenerator.inflate(rng.next()))
+                       .insert(i);
+            }
+
+            for (int i = 0; i < partitionSize; i++)
+            {
+                history.visitPartition(1)
+                       .deleteRow(i);
+
+                visitor.replayAll();
+                history.validateAll(tracker, sut);
+            }
+        }
+    }
+
+
+    @Test
+    @SuppressWarnings("rawtypes")
+    public void regularAndStaticOverrideTest() throws Throwable
+    {
+        for (ColumnSpec.DataType type : new ColumnSpec.DataType[]{ ColumnSpec.asciiType(2, 0), ColumnSpec.int64Type })
+        {
+            SchemaSpec schema = new SchemaSpec("harry",
+                                               "test_overrides",
+                                               Arrays.asList(ColumnSpec.pk("pk1", ColumnSpec.asciiType(4, 10))),
+                                               Arrays.asList(ColumnSpec.ck("ck1", type, false)),
+                                               Arrays.asList(ColumnSpec.regularColumn("regular1", ColumnSpec.asciiType(2, 2)),
+                                                             ColumnSpec.regularColumn("regular2", ColumnSpec.int64Type)),
+                                               Arrays.asList(ColumnSpec.staticColumn("static1", ColumnSpec.asciiType(2, 2)),
+                                                             ColumnSpec.staticColumn("static2", ColumnSpec.int64Type)
+                                               ));
+
+            Map<String, Bijections.Bijection<?>> reGenerators = new HashMap<>();
+            reGenerators.put("regular1", ColumnSpec.asciiType(4, 4).generator());
+            reGenerators.put("regular2", ColumnSpec.int64Type.generator());
+            reGenerators.put("static1", ColumnSpec.asciiType(8, 4).generator());
+            reGenerators.put("static2", ColumnSpec.int64Type.generator());
+
+            DataTracker tracker = new DefaultDataTracker();
+            beforeEach();
+            sut.schemaChange(schema.compile().cql());
+
+            TokenPlacementModel.ReplicationFactor rf = new TokenPlacementModel.SimpleReplicationFactor(1);
+
+            int partitionSize = 100;
+            HistoryBuilder history = new HistoryBuilder(seed, partitionSize, 10, schema, rf);
+            ReplayingVisitor visitor = history.visitor(tracker, sut, SystemUnderTest.ConsistencyLevel.ALL);
+            EntropySource rng = new JdkRandomEntropySource(seed);
+
+            Map<String, Set<Object>> perColumnOverrides = new HashMap<>();
+            for (ColumnSpec column : schema.regularColumns)
+            {
+                perColumnOverrides.put(column.name, new HashSet<>());
+                for (int i = 0; i < partitionSize; i++)
+                {
+                    Object override = reGenerators.get(column.name).inflate(rng.next());
+                    history.valueOverrides().override(column.name, i, override);
```

Comment:
So we're overriding the value at every row of a 100-row partition?

Reply:
You are right to note that using `partitionSize` here is somewhat arbitrary. In other words, there's nothing prevents us to use some other `idx` (i.e. larger than 100), and get an actually random value. I've just overriden 100 values per column, but then we're going in the loop and are simply picking one of these 100 values. In the end, there will be only 1 unique static value per partition and ~60 unique values (assuming uniform distribution, so can be more or less) per value column.